docs/labwc-config.5.scd

```diff
@@ -330,6 +330,50 @@ The rest of this man page describes configuration options.
 	any motion events while a keyboard is typing, and for a short while
 	after as well.
 
+## WINDOW RULES
+
+Two types of window rules are supported, actions and properties. They are
+defined as shown below. Property values can be *yes*, *no* or *unset*.
+
+```
+<windowRules>
+
+  <!-- Action -->
+  <windowRule identifier="" event="">
+    <action name="">
+  </windowRules>
+
+  <!-- Property -->
+  <windowRule identifier="">
+    <serverDecoration></serverDecoration>
+  </windowRules>
+
+</windowRules>
+```
+
+If a window matches criteria for multiple rules which set the same property,
+later config entires have higher priority. *unset* can be useful in this
+situation.
+
+*<windowRules><windowRule identifier="" event="">*
+	Define a window rule for any window which matches the criteria defined
+	by the attribute *identifier*. Matching against patterns with '\*'
+	(wildcard) and '?' (joker) is supported. Pattern matching is
+	case-insensitive.
+
+	*identifier* relates to app_id for native Wayland windows and WM_CLASS
+	for XWayland clients.
+
+	*event* is only required for window-rules with *<action>* and defines
+	the event that the rule relates to.  The following events are
+	supported:
+
+		- *onMap*
```

Comment:
Could there be any more events we want to react to?
If there are none I think we could simply drop the `event` attribute and always execute the actions (if defined) on map.

Reply:
Not sure.

I just wanted to avoid breaking changes later, but I suppose we could just default to 'onMap'.

Sway only apply window-rules when windows 'appear' - so on the basis that sway users have not required anything different, we're probably safe :smile:

Wayfire supports the following events: created, unmaximized, maximized, minimized and fullscreened
I currently can't think of use-cases for these.